### PR TITLE
Docs: Cascade Deletes

### PR DESCRIPTION
**Pull request by gregnr**
Adds docs on the different types of foreign key deletes.

https://docs-git-docs-cascade-deletes-supabase.vercel.app/docs/guides/database/postgres/cascade-deletes

Also adds link from dashboard:
image

### CODE REVIEW

File: apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts [hunk 1]
```diff
@@ -471,6 +471,10 @@ export const database = {
           name: 'Managing Indexes',
           url: '/guides/database/postgres/indexes',
         },
+        {
+          name: 'On Delete Constraint Behavior',
```
Comment (kiwicopple) [inline apps/docs/components/Navigation/NavigationMenu/NavigationMenu.constants.ts:475]:
```suggestion
          name: 'Cascading Deletes',
```

File: apps/docs/pages/guides/database/postgres/on-delete-constraint-behavior.mdx [hunk 2]
```diff
@@ -0,0 +1,198 @@
+import Layout from '~/layouts/DefaultGuideLayout'
+
+export const meta = {
+  title: 'On Delete Constraint Behavior',
```
Comment (kiwicopple) [inline apps/docs/pages/guides/database/postgres/on-delete-constraint-behavior.mdx:4]:
```suggestion
  title: 'Cascade Deletes',
```

File: apps/docs/pages/guides/database/postgres/on-delete-constraint-behavior.mdx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+3. **SET NULL:** When a row is deleted from the parent table, the values of the foreign key columns in the child table(s) are set to NULL.
+4. **SET DEFAULT:** When a row is deleted from the parent table, the values of the foreign key columns in the child table(s) are set to their default values.
+5. **NO ACTION:** This option is similar to RESTRICT, but it also has the option to be “deferred” to the end of a transaction. This means that other cascading deletes can run first, and then this delete constraint will only throw an error if there is referenced data remaining _at the end of the transaction_.
+
+These options can be specified when defining a foreign key constraint using the "ON DELETE" clause. For example, the following SQL statement creates a foreign key constraint with the `CASCADE` option:
+
+```sql
+alter table
+  child_table add constraint fk_parent foreign key (parent_id) references parent_table (id) on delete cascade;
+```
```
Comment (kiwicopple) [inline apps/docs/pages/guides/database/postgres/on-delete-constraint-behavior.mdx:22]: is this prettier formatting it this way? Frustrating, I thought we had fixed it

**Review by kiwicopple — APPROVED** [review]

File: apps/docs/pages/guides/database/postgres/on-delete-constraint-behavior.mdx (hunk 2 above)
Comment (gregnr) [reply apps/docs/pages/guides/database/postgres/on-delete-constraint-behavior.mdx]: 👍 there were a few others I had to fix, did it all in one commit: https://github.com/supabase/supabase/pull/13972/commits/c17fe90603ca2581fc5b91a6aac9fd576ce1f7de

File: apps/docs/pages/guides/database/postgres/on-delete-constraint-behavior.mdx (hunk 3 above)
Comment (gregnr) [reply apps/docs/pages/guides/database/postgres/on-delete-constraint-behavior.mdx]: Yeah looks like it. Sorry didn't realize there was a history here! I imagine ideally this is wrapping sooner, right?

Comment (gregnr) [reply apps/docs/pages/guides/database/postgres/on-delete-constraint-behavior.mdx]: Figured it out - 2 issues:
1. `prettier-plugin-sql-cst` wasn't working - I believe it was getting overridden by `prettier-plugin-sql` which is a sub dep of `@supabase/postgres-meta`. I fixed this by disabling that plugin and explicitly enabling this one.
2. Turns out `prettier-plugin-sql-cst` only understands SQLite, so any time it encounters Postgres-specific code that SQLite can't do (eg. dropping constraints), it fails to format. Not high priority issue as worst case we can just format manually - may not be solvable right now. Tracking here anyway:
  https://github.com/supabase/supabase/issues/14019

**Comment by gregnr** [comment]
Ah prettier CI is now failing. Looking into it.

**Pull request merged**